examples/ui/feathers.rs

```diff
@@ -0,0 +1,180 @@
+//! This example shows off the various Bevy Feathers widgets.
+
+use bevy::{
+    core_widgets::{CoreWidgetsPlugin, SliderStep},
+    feathers::{
+        controls::{button, slider, ButtonProps, ButtonVariant, SliderProps},
+        dark::create_dark_theme,
+        theme::{self, corners::RoundedCorners, ThemeBackgroundColor, UiTheme, UseTheme},
+        FeathersPlugin,
+    },
+    input_focus::{
+        tab_navigation::{TabGroup, TabNavigationPlugin},
+        InputDispatchPlugin,
+    },
+    prelude::*,
+    ui::InteractionDisabled,
+    winit::WinitSettings,
+};
+
+fn main() {
+    App::new()
+        .add_plugins((
+            DefaultPlugins,
+            CoreWidgetsPlugin,
+            InputDispatchPlugin,
+            TabNavigationPlugin,
+            FeathersPlugin,
+        ))
+        .insert_resource(UiTheme(create_dark_theme()))
+        // Only run the app when there is user input. This will significantly reduce CPU/GPU use.
+        .insert_resource(WinitSettings::desktop_app())
+        .add_systems(Startup, setup)
+        .run();
+}
+
+fn setup(mut commands: Commands) {
+    // ui camera
+    commands.spawn(Camera2d);
+    let root = demo_root(&mut commands);
+    commands.spawn(root);
+}
+
+fn demo_root(commands: &mut Commands) -> impl Bundle {
+    (
+        Node {
+            width: Val::Percent(100.0),
+            height: Val::Percent(100.0),
+            align_items: AlignItems::Start,
+            justify_content: JustifyContent::Start,
+            display: Display::Flex,
+            flex_direction: FlexDirection::Column,
+            row_gap: Val::Px(10.0),
+            ..default()
+        },
+        TabGroup::default(),
+        ThemeBackgroundColor(theme::tokens::WINDOW_BG),
+        children![(
+            Node {
+                display: Display::Flex,
+                flex_direction: FlexDirection::Column,
+                align_items: AlignItems::Stretch,
+                justify_content: JustifyContent::Start,
+                padding: UiRect::all(Val::Px(8.0)),
+                row_gap: Val::Px(8.0),
+                width: Val::Percent(30.),
+                min_width: Val::Px(200.),
+                ..default()
+            },
+            children![
+                (
+                    Node {
+                        display: Display::Flex,
+                        flex_direction: FlexDirection::Row,
+                        align_items: AlignItems::Center,
+                        justify_content: JustifyContent::Start,
+                        column_gap: Val::Px(8.0),
+                        ..default()
+                    },
+                    children![
+                        button(
+                            ButtonProps {
+                                on_click: Some(commands.register_system(|| {
```

Comment:
I haven't kept up with all the various ecs and bsn work. So out of curisoity, is there a future where we'll be able to do something like this:

```rust
ButtonProps {
	click_handler: || { 
		info!("clicked!");
	},
	..default()
}
```

Reply:
Yes, I would like to be able to do that. I need to take a proper crack at the code here though.